VisualRust/Racer/RacerSingleton.cs

```diff
@@ -67,10 +71,53 @@ private void ReinitializeRacerPaths()
             else
                 racerPathForExecution = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "Racer", BundledRacerExecutable);
             // Same for custom RUST_SRC_PATH
-            if(GetVisualRustProperty<bool>(env, nameof(RustOptionsPage.UseCustomSources)))
-                racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
-            else
-                racerSourcesLocation = null;
+            var type = GetVisualRustProperty<RustOptionsPage.RustSource>(env, nameof(RustOptionsPage.SourceType));
+            switch (type) {
+                case RustOptionsPage.RustSource.EnvVariable:
+                    racerSourcesLocation = null;
+                    break;
+                case RustOptionsPage.RustSource.Custom:
+                    racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
+                    break;
+                default:
+                    racerSourcesLocation = GetRustcSysroot();
+                    if (racerSourcesLocation != null && !Directory.Exists(racerSourcesLocation))
+                        racerSourcesLocation = null;
+                    break;
+            }
+
+        }
+
+        private static String GetRustcSysroot()
```

Comment:
Have you tried to use the already existing [`GetSysrootAsync()`](https://github.com/PistonDevelopers/VisualRust/blob/e984dfebd4a8136b83d4dcc241011579102e7042/VisualRust.Shared/Cargo.cs#L64-L68) method that I mentioned in the issue? Probably it's a bit hard to use here directly, because it requires an `async` context. I think the whole of `Racer.Exec()` should be async, but I can file that as a follow-up issue if you aren't able to do this now. In that case, please add a `TODO` comment here to eventually switch to using `Cargo.GetSysrootAsync()` .